kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualClusterModel.java

```diff
@@ -374,6 +374,11 @@ public Optional<Tls> getTls() {
 
         private Optional<SslContext> buildDownstreamSslContext() {
             return tls.map(tlsConfiguration -> {
+                if (tlsConfiguration.key() == null) {
+                    throw new IllegalConfigurationException(("Gateway '%s' of virtual cluster '%s' has a `tls' object defined with no 'key' configured. "
```

Comment:
A link off to the docs like @SamBarker is toying with in #2422 might be good, if we can land on a sustainable enough way to link from code to docs.